frontend/src/components/Species/SpeciesDetails.tsx

```diff
@@ -49,7 +49,8 @@ export const SpeciesDetails = () => {
   }
 
   const onWrite = async (editData: EditDataType<SpeciesDetailsType>) => {
-    await editSpeciesRequest(editData)
+    const { species_id } = await editSpeciesRequest(editData).unwrap()
+    setTimeout(() => navigate(`/species/${species_id}`), 15)
```

Comment:
This is not the prettiest fix as mentioned but it is simple so it is acceptable.